bookshelf/test/lib/E2EDeploymentTrait.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Samples\Bookshelf;
+
+use Symfony\Component\Yaml\Dumper;
+
+/**
+ * Class E2EDeploymentTrait
+ * @package Google\Cloud\Samples\Bookshelf
+ *
+ * Use this trait to deploy the project to GCP for an end-to-end test.
+ */
+trait E2EDeploymentTrait
+{
+    use GetConfigTrait;
+    use SkipTestsIfMissingCredentialsTrait;
+
+    /** @staticvar array $versions */
+    public static $versions;
+
+    /** @staticvar Gcloud $gcloud */
+    public static $gcloud;
+
+    private static function dumpConfig()
+    {
+        $config = self::getConfig();
+        $dumper = new Dumper();
+        $yaml = $dumper->dump($config);
+        // TODO: Use different filename
+        file_put_contents(__DIR__ . '/../../config/settings-e2e.yml', $yaml);
```

Comment:
Keeping the settings separate seems like a good idea, although it's kinda odd to do this since we're using the same credentials in both places anyway.